### PR TITLE
bpo-41659: Disallow curly brace directly after primary

### PR DESCRIPTION
**Pull request by lysnikolaou**
https://bugs.python.org/issue41659

### CODE REVIEW

File: Grammar/python.gram [hunk 1]
```diff
@@ -475,6 +475,7 @@ await_primary[expr_ty] (memo):
     | AWAIT a=primary { CHECK_VERSION(expr_ty, 5, "Await expressions are", _Py_Await(a, EXTRA)) }
     | primary
 primary[expr_ty]:
+    | invalid_primary
```
Comment (gvanrossum) [inline Grammar/python.gram]: I suppose there's a reason the error alternative comes first? Is it because this rule is so recursive?

Comment (lysnikolaou) [reply Grammar/python.gram]: It doesn't have to be first, but it does have to be before the `primary genexp` alternative. There's a thorough explanation of why this is on the issue. Should I add a comment about that in the grammar?

Comment (gvanrossum) [reply Grammar/python.gram]: Maybe keep it where it is and add a brief comment "must be before 'primay genexp' because of invalid_genexp"?

Comment (lysnikolaou) [reply Grammar/python.gram]: Done in https://github.com/python/cpython/pull/22996/commits/aada2f0abe0a823366e9a33b2826d83b1c746e0f.

**Review by gvanrossum — APPROVED** [review]
Please add a news item. Also I think this deserves backporting.

**Review by pablogsal — COMMENTED** [review]
Could you also add a test to exercise this code path (or check that the error happens where we expect)?

**Pull request merged**